### PR TITLE
samples: dect_phy: dect_shell: mac updates

### PR DESCRIPTION
**Pull request by jhirsi**
This PR brings following new functionalities:
- PT/client side when associated: periodic background scanning towards associated beacon device to keep up-to-date timing information in neighbor list.
- Bi-directional data transfer between two beacon devices that have each other as a neighbor (i.e. scanned).

### CODE REVIEW

File: samples/dect/dect_phy/dect_shell/README.rst [hunk 1]
```diff
@@ -895,6 +932,92 @@ Example: starting of cluster beacon and sending RA data to it
...
+      desh:~$ dect mac beacon_scan -c 1675 -f
+
+* FT/Beacon device 1 - Send association request to device 2:
+
+  .. code-block:: console
+
+      desh:~$ dect mac associate -t 2
+
+
+* FT/Beacon device 2 - Send association request to device 2:
```
Comment (peknis) [inline samples/dect/dect_phy/dect_shell/README.rst:979]:
```suggestion
* FT/Beacon device 2 - Send association request to device 1:
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_shell.c [hunk 2]
```diff
@@ -418,8 +419,9 @@ static int dect_phy_mac_dissociate_cmd(const struct shell *shell, size_t argc, c
 
 	ret = dect_phy_mac_ctrl_dissociate(&params);
 	if (ret) {
-		desh_error("Cannot send association to FT %u's random access resource, err %d",
-			   params.target_long_rd_id, ret);
+		desh_error("Cannot send association release to FT %u's random access resource, "
+			   "err %d",
+				params.target_long_rd_id, ret);
```
Comment (trantanen) [inline samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_shell.c:424]: Is indentation fine in real life and the view makes it look weird?

File: samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+		MODEM_TICKS_TO_MS(time_now - bg_scan_data->last_updated_rcv_time);
+
+	desh_print("     Running: true");
+	desh_print("     Phy op handle:  %u", bg_scan_data->params.phy_op_handle);
+
+	if (bg_scan_data->last_updated_rcv_time == 0) {
+		desh_print("     Last seen: never");
+	} else {
+		desh_print("     Last seen:      %d msecs ago",
+				time_from_last_received_ms);
```
Comment (trantanen) [inline samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c:297]: Should this be lld?
```suggestion
		desh_print("     Last seen:      %lld msecs ago",
				time_from_last_received_ms);
```

File: samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+		desh_print("     Last timestamp: %lld mdm ticks",
+				bg_scan_data->last_updated_rcv_time);
```
Comment (trantanen) [inline samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c:299]: Should this be llu?
```suggestion
		desh_print("     Last timestamp: %llu mdm ticks",
				bg_scan_data->last_updated_rcv_time);
```

File: samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+	uint32_t scan_info_time_shift_updated_count;
+	int64_t scan_info_time_shift_last_value;
+};
+
+struct dect_phy_mac_nbr_bg_scan_data {
+	bool running;
+	struct dect_phy_mac_nbr_bg_scan_params params;
+	struct dect_phy_mac_nbr_bg_scan_metrics_data metrics;
+
+	uint64_t last_updated_rcv_time;
```
Comment (trantanen) [inline samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c:43]: If this is in milliseconds, Zephyr "standard" would be to use `int64_t` but I guess `uint64_t` is pretty much built into DESH already.

File: samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+	if (bg_scan_data->last_updated_rcv_time != 0 &&
+	    time_from_last_received_ms > DECT_PHY_MAC_NBR_BG_SCAN_MAX_UNREACHABLE_TIME_MS) {
+		bg_scan_data->running = false;
+		dect_phy_api_scheduler_list_item_remove_by_phy_op_handle(params->handle);
+		if (bg_scan_data->params.cb_op_completed) {
+			bg_scan_data->params.cb_op_completed(&(
+				struct dect_phy_mac_nbr_bg_scan_op_completed_info){
+				.cause = DECT_PHY_MAC_NBR_BG_SCAN_OP_COMPLETED_CAUSE_UNREACHABLE,
+				.phy_op_handle = params->handle,
+				.target_long_rd_id = bg_scan_data->params.target_long_rd_id,
```
Comment (trantanen) [inline samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c:113]: I don't know if this way of passing a structure was used elsewhere but I guess we could introduce it as a parameter before the call and pass it in a bit more normal manner?

File: samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c (hunk 5 above)
Comment (jhirsi) [reply samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c:43]: This is in modem ticks. Maybe I could rename the variable accordingly.

File: samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_client.c [hunk 7]
```diff
@@ -627,6 +651,163 @@ int dect_phy_mac_client_associate(struct dect_phy_mac_nbr_info_list_item *target
...
+		(__func__), common_header->transmitter_id);
+
+	struct dect_phy_mac_nbr_bg_scan_params bg_scan_params;
+
+	bg_scan_params.cb_op_completed = dect_phy_mac_client_nbr_scan_completed_cb;
+	bg_scan_params.target_nbr = association_data->target_nbr;
+	bg_scan_params.target_long_rd_id = association_data->target_long_rd_id;
+	bg_scan_params.phy_op_handle = association_data->bg_scan_phy_handle;
+
+	if (association_data->bg_scan_on_going) {
```
Comment (tokangas) [inline samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_client.c:833]: A stupid question, why is background scan first stopped and then started immediately?

**Review by peknis — APPROVED** [review]

File: samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c (hunk 6 above)
Comment (jhirsi) [reply samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_nbr_bg_scan.c:113]: I was surprised that this worked with C-language like this. OK, I'll change these. from this file.

File: samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_client.c (hunk 7 above)
Comment (jhirsi) [reply samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_client.c:833]: Yes

**Review by nordicjm — APPROVED** [review]

File: samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_shell.c (hunk 2 above)
Comment (jhirsi) [reply samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_shell.c:424]: For me it looks ok in code.

**Review by tokangas — APPROVED** [review]

**Review by trantanen — APPROVED** [review]

**Pull request merged**